docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,88 @@
+# Refactor Subscription Platform frontend
+
+- Status: proposed
+- Deciders: Ben Bangert, Lisa Chan
+- Date: 2023-03-07
+
+## Context and Problem Statement
+
+The Subscription Platform frontend has become overly complex, making it inflexible and difficult to work with. This has resulted in longer development times on new features and in some cases left us unable to easily accommodate requests from relying parties.
+
+Additionally, as the app has grown, so have the number of network calls causing a reduction in app performance. The aim is to refactor the app to have one network call for each action.
+
+Finally, the frontend includes some business and application logic, which has made components and parts of the app unnecessarily complex. As a result the frontend is less flexible, and harder to change and maintain. To solve this, the Subscription Platform team wants to move all business and application logic to the backend.
+
+To address the problems briefly discussed above, and to meet current and future relying party and feature requests, refactoring of the existing Subscription Platform is necessary.
+
+## Decision Drivers
+
+- Faster development - This applies to both the refactoring necessary for SubPlat 3.0, as well as future relying party and feature requests
+- Easier routing - The SubPlat teams wants to perform more advanced routing to allow features such as, Relying Party specific checkout pages, A/B testing, etc,
+- GraphQL support - Improve app performance and responsivness by reducing the number of network requests
+- Reuse existing components - We have received generally favourable reviews of our existing frontend look and feel, and want to keep the appearance the same
+
+## Considered Options
+
+- A. React Framework - Next.js
+- B. React Framework - Gatsby.js
+- C. React, Typescript, without Redux
+- D. React, Typescript, with Redux
```

Comment:
Remix is the new hotness I think.  Would you consider it?

Reply:
Oh interesting, the team actually hadn't considered Remix. I'll bring it up in our team meeting later today. Great suggestion thank you.

Here are some of my initial thoughts.
- Looks really slick. Especially because of this comment from Reddit 
  - "...[my mind was blown away by how smooth and relaxing my development experience became](https://www.reddit.com/r/reactjs/comments/112bujy/switched_from_nextjs_to_remixjs_and_loving_it/)..."
- We are currently probably looking at an SSR app, and if performance is our main driver, the comment below makes me think maybe there won't be as big a performance difference?
  - "[If the Next.js app moved away from client fetching, and used getServerSideProps, they would probably close the gap and have simpler answers to these questions.](https://remix.run/blog/remix-vs-next)..."
- The [drawbacks mentioned here](https://blog.logrocket.com/guide-to-remix-react-framework/#drawbacks-remix) as well as the [conclusion from this piece](https://dev.to/braincuber_technologies/remix-vs-nextjs-a-detailed-comparison-2ogg), are probably my main concerns.
- Do you happen to know if Remix is used by any other teams in Mozilla? I know Next.js is.

Reply:
> Do you happen to know if Remix is used by any other teams in Mozilla? I know Next.js is.

No.  I don't know what other teams use.

Reply:
Just to circle back to this. We discussed this as a team, and decided to stick with Next. Along with the drawbacks mentioned above, we primarily favor Next for its popularity, community size, proven production use, and that its already used by other teams at Mozilla.

Good suggestion though, and I'll update the docs with the option.